src/main/java/com/epam/grid/engine/provider/job/slurm/SlurmJobProvider.java

```diff
@@ -134,6 +158,93 @@ public InputStream getJobLogFile(final int jobId, final JobLogInfo.Type logType)
         throw new UnsupportedOperationException("Job log info file retrieving operation haven't implemented yet");
     }
 
+    /**
+     * Creates the structure of an executable command based on the passed options.
+     *
+     * @param options User-defined options.
+     * @return The structure of an executable command.
+     */
+    private String[] makeSbatchCommand(final JobOptions options) {
+        final Context context = new Context();
+        context.setVariable(OPTIONS, options);
+        context.setVariable(LOG_DIR, logDir);
+        context.setVariable(ENV_VARIABLES, extractEnvVariablesFromOptions(options));
+        return commandCompiler.compileCommand(getProviderType(), SBATCH_COMMAND, context);
+    }
+
+    private void validateJobOptions(final JobOptions options) {
+        if (!StringUtils.hasText(options.getCommand())) {
+            throw new GridEngineException(HttpStatus.BAD_REQUEST, "Command should be specified!");
+        }
+        checkPriorityRange(options);
+        if (options.getParallelEnvOptions() != null) {
+            throw new UnsupportedOperationException("Parallel environment variables are not supported yet!");
+        }
+        if (options.isCanBeBinary()) {
+            throw new UnsupportedOperationException("Scripts from command line are not supported yet!");
+        }
+    }
+
+    private void checkPriorityRange(final JobOptions options) {
+        Optional.of(options.getPriority()).stream()
```

Comment:
1. Do we really need to convert to `Stream`? Seems, like
```java
Optional.of().filter().ifPresent()
```
will do the same.

2. Since priority is a primitive, we can skip using `Optional` at all:
```java
if (options.getPriority() < 0) {
    throw ...
}
```

Reply:
fixed